--- a/libavcodec/mpegvideo_enc.c
+++ b/libavcodec/mpegvideo_enc.c
@@ -1437,7 +1437,7 @@
 }

 int ff_MPV_encode_picture(AVCodecContext *avctx, AVPacket *pkt,
-                          AVFrame *pic_arg, int *got_packet)
+                          const AVFrame *pic_arg, int *got_packet)
 {
     MpegEncContext *s = avctx->priv_data;
     int i, stuffing_count, ret;

--- a/libavcodec/mpegvideo.h
+++ b/libavcodec/mpegvideo.h
@@ -803,7 +803,7 @@
 int ff_MPV_encode_init(AVCodecContext *avctx);
 int ff_MPV_encode_end(AVCodecContext *avctx);
 int ff_MPV_encode_picture(AVCodecContext *avctx, AVPacket *pkt,
-                          AVFrame *frame, int *got_packet);
+                          const AVFrame *frame, int *got_packet);
 void ff_dct_encode_init_x86(MpegEncContext *s);
 void ff_MPV_common_init_x86(MpegEncContext *s);
 void ff_MPV_common_init_axp(MpegEncContext *s);

--- a/libavcodec/qpeg.c
+++ b/libavcodec/qpeg.c
@@ -325,8 +325,8 @@
 static av_cold int decode_init(AVCodecContext *avctx){
     QpegContext * const a = avctx->priv_data;

-    avcodec_get_frame_defaults(&a->pic);
-    avcodec_get_frame_defaults(&a->ref);
+    avcodec_get_frame_defaults(a->pic);
+    avcodec_get_frame_defaults(a->ref);
     a->avctx = avctx;
     avctx->pix_fmt= AV_PIX_FMT_PAL8;

--- a/libavcodec/snowenc.c
+++ b/libavcodec/snowenc.c
@@ -1549,7 +1549,7 @@
 }

 static int encode_frame(AVCodecContext *avctx, AVPacket *pkt,
-                        AVFrame *pict, int *got_packet)
+                        const AVFrame *pict, int *got_packet)
 {
     SnowContext *s = avctx->priv_data;
     RangeCoder * const c= &s->c;

--- a/libavcodec/utils.c
+++ b/libavcodec/utils.c
@@ -976,7 +976,7 @@
         return ret;
     }

-    av_image_copy(frame->data, frame->linesize, tmp.data, tmp.linesize,
+    av_image_copy(frame->data, frame->linesize, (const uint8_t **)tmp.data, tmp.linesize,
                   frame->format, frame->width, frame->height);

     av_frame_unref(&tmp);

--- a/libavutil/frame.c
+++ b/libavutil/frame.c
@@ -271,7 +271,7 @@
             av_samples_copy(dst->extended_data, src->extended_data, 0, 0,
                             dst->nb_samples, ch, dst->format);
         } else {
-            av_image_copy(dst->data, dst->linesize, src->data, src->linesize,
+            av_image_copy(dst->data, dst->linesize, (const uint8_t **)src->data, src->linesize,
                           dst->format, dst->width, dst->height);
         }
         return 0;
@@ -424,7 +424,7 @@
         av_samples_copy(tmp.extended_data, frame->extended_data, 0, 0,
                         frame->nb_samples, ch, frame->format);
     } else {
-        av_image_copy(tmp.data, tmp.linesize, frame->data, frame->linesize,
+        av_image_copy(tmp.data, tmp.linesize, (const uint8_t **)frame->data, frame->linesize,
                       frame->format, frame->width, frame->height);
     }


